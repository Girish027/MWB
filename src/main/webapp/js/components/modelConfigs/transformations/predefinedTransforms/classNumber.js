import Constants from 'constants/Constants';

const classNumber = () => {
  const regex = {
    type: Constants.TRANSFORMATION_TYPES.WORDCLASS_SUBST_REGEX,
    mappings: {},

  };
  regex.mappings = {
    '/\\b[0-9]+(?:\\s*[\\.,]\\s*[0-9]+)*(?:\\s*(?:nd|th|rd|st))?\\b/i': '_class_number',
    '/\\b(?:zero|one|two|three|four|five|six|seven|eight|nine|hundred|ten|eleven|twelve|thirteen|fourteen|fifteen|sixteen|seventeen|eighteen|nineteen|twenty|thirty|forty|fifty|sixty|seventy|eighty|ninety)\'?s\\b/i': '_class_number',
    '/#(?=[^a-zA-Z0-9])/i': '_class_number_ref',
  };

  return regex;
};

export default classNumber;
