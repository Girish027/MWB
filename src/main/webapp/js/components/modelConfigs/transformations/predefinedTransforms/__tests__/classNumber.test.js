import classNumber from 'components/modelConfigs/transformations/predefinedTransforms/classNumber';

describe('classNumber', () => {
  const regex = classNumber();

  test('type is equal to wordclass-subst-regex', () => {
    expect(regex.type).toEqual('wordclass-subst-regex');
  });

  test('mapping contains more than one key', () => {
    const keys = Object.keys(regex.mappings);
    expect(keys.length).toBeGreaterThan(1);
  });

  test('snapshot predefined transformation', () => {
    expect(regex.mappings).toMatchSnapshot();
  });
});
