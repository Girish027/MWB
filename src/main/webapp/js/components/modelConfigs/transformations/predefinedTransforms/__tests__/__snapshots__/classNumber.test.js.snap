// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`classNumber snapshot predefined transformation 1`] = `
Object {
  "/#(?=[^a-zA-Z0-9])/i": "_class_number_ref",
  "/\\\\b(?:zero|one|two|three|four|five|six|seven|eight|nine|hundred|ten|eleven|twelve|thirteen|fourteen|fifteen|sixteen|seventeen|eighteen|nineteen|twenty|thirty|forty|fifty|sixty|seventy|eighty|ninety)'?s\\\\b/i": "_class_number",
  "/\\\\b[0-9]+(?:\\\\s*[\\\\.,]\\\\s*[0-9]+)*(?:\\\\s*(?:nd|th|rd|st))?\\\\b/i": "_class_number",
}
`;
